### PR TITLE
Improve ordering and add payment dialogue

### PR DESCRIPTION
**Pull request by ec-m**
I increased the specificity of ice cream order state such that general conversations about ice cream are ignored. Furthermore, the basic framework of the payment dialogue has been added and the `__init__.py` is changed such that all newly implemented modules do not have to be added by hand but are automatically included with `mod.add(..)`. I think it might make sense to merge this such that if other people start working on their features for the week, they could already make use of the stuff I included.

### CODE REVIEW

**Comment by josephbirkner** [comment]
Please make changes for #1 first, then this is a bit easier to review.

**Comment by josephbirkner** [comment]
Please resolve conflicts... ❌

**Comment by ec-m** [comment]
> Please resolve conflicts... ❌

Done. I hope all conflicts are solved now :)

**Review by josephbirkner — COMMENTED** [review]
Generally nice implementation! 🌈 🌞 One nitpick that will lead to errors at some point: You are using the `Property.read` and `Property.write` functions directly a LOT. This is not supposed to work (see this Ravestate issue: https://github.com/Roboy/ravestate/issues/40 at §5). The problem with using the functions directly is that __(a)__ it is not thread-safe (remember threads are executed in parallel), and __(b)__ it prevents multiple contexts from using the property simultaneously (technically ravestate is not limited to one context per process).

TL;DR: Please access properties through `ctx[prop]` or `ctx[prop] = y` with access declarations in the states 😊

**Comment by ec-m** [comment]
> TL;DR: Please access properties through `ctx[prop]` or `ctx[prop] = y` with access declarations in the states 😊

I am not sure what you mean by writing access declaration in the states? Can you give me a pointer to examples where this kind of accessing of properties is implemented. That would be great, thank you!

**Comment by ec-m** [comment]
> TL;DR: Please access properties through `ctx[prop]` or `ctx[prop] = y` with access declarations in the states 😊

Done (together with changing the architecture such that this is possible).

**Review by josephbirkner — APPROVED** [review]
👍
